scipy/stats/_continuous_distns.py

```diff
@@ -264,6 +264,9 @@ def fit(self, data, **kwds):
 
         data = np.asarray(data)
 
+        if np.logical_not(np.isfinite(data)).any():
```

Comment:
Drive-by-comment: `not np.isfinite(data).all()` is likely faster since `logical_not` generates an intermediate array.

Reply:
> Drive-by-comment: `not np.isfinite(data).all()` is likely faster since `logical_not` generates an intermediate array.

I bet this matters more on larger arrays. I was surprised to learn even on the test case there is a speedup: 
```
import scipy.stats as stats
import numpy as np
from datetime import datetime
startTime = datetime.now()
for i in  range(0, 10000):
    x = np.array([1.6483, 2.7169, 2.4667, 1.1791, 3.5433, np.nan])          
    np.testing.assert_raises(RuntimeError, stats.norm.fit, x)

print(datetime.now() - startTime)
```
gives runtime of: 
# 0:00:00.156653 (old) 
# 0:00:00.138408 (proposed)

for a larger range 10000 -> 1000000 the improvements are: 
0:00:12.887804 (old) 
0:00:12.201071 (new)   
 
I'll change all the implementations to that proposed by @pv  Thanks!